src/vec_int128_ppc.h

```diff
@@ -1949,6 +2210,105 @@ vec_minuq(vui128_t vra, vui128_t vrb)
   return (vui128_t) vec_sel ((vui32_t) vrb, (vui32_t) vra, minmask);
 }
 
+/** \brief Vector Modulo by const 10e31 Signed Quadword.
+ *
+ *  Compute the remainder of a 128 bit values vra % 10e31.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  8-52 | 1/cycle  |
+ *  |power9   |  9-23 | 2/cycle  |
+ *
+ *  @param vra the dividend as a vector treated as a signed __int128.
+ *  @param q 128-bit signed __int128 containing the quotient from vec_divuq_10e31().
+ *  @return the remainder as vector signed __int128.
+ */
+static inline vi128_t
+vec_modsq_10e31 (vi128_t vra, vi128_t q)
+{
+  const vui128_t zero = (vui128_t) vec_splats ((int) 0);
```

Comment:
probably compiles equivalently, but can you just use a 0 initializer here?  `{0}`?
Since this code serves as example code for more general purposes, I'm looking for ways to make it simpler.
This construction appears in other places as well.

Reply:
Because I want a single instruction splat immediate not a load vector. And I have learned the hard way, not to blindly trust the compiler.

Reply:
Actually is it worse. Constant zero is very common and I really want the compiler to common up the various const zeros and generate it only once. But the compiler will not if they are created with different element sizes :( 

So I have been trying to standardize on splat word immediate for word and larger elements. I am still trying to figure out what the various compiler versions do with:

const vui128_t zero = (vui128_t) { (__int128) 0 };

guess I have to create another dummy function ....

Reply:
looks like the compiler generates vspltisw for this case. But I am not seeing the compiler do much commoning for the const zero. In fact the examples I have show the compiler generating the vspltisw 0 four times, Not sure if the compiler just not seeing the opportunity or is avoiding register pressure ( a valid concern) 
I need to think about this.

Reply:
Ya looks like vspltisw is so cheap that regenerating is usually better then dealing with register pressure and spill.

So I summit this change for div/mod operations but not ready to do this in general.